parts/linux/cloud-init/nodecustomdata.yml

```diff
@@ -196,6 +196,45 @@ write_files:
     APT::Periodic::Unattended-Upgrade "0";
 {{end}}
 
+{{- if ShouldConfigureHTTPProxy}}
+- path: /etc/apt/apt.conf.d/95proxy
+  permissions: "0644"
+  owner: root
+  content: |
+    {{- if HasHTTPProxy }}
+    Acquire::http::proxy "{{GetHTTPProxy}}";
+    {{- end}}
+    {{- if HasHTTPSProxy }}
+    Acquire::https::proxy "{{GetHTTPSProxy}}";
+    {{- end}}
+
+- path: /etc/systemd/system.conf.d/proxy.conf
+  permissions: "0644"
+  owner: root
+  content: |
+    [Manager]
+    {{- if HasHTTPProxy }}
+    DefaultEnvironment="HTTP_PROXY={{GetHTTPProxy}}"
+    DefaultEnvironment="http_proxy={{GetHTTPProxy}}"
+    {{- end}}
+    {{- if HasHTTPSProxy }}
+    DefaultEnvironment="HTTPS_PROXY={{GetHTTPSProxy}}"
+    DefaultEnvironment="https_proxy={{GetHTTPSProxy}}"
+    {{- end}}
+    {{- if HasNoProxy }}
+    DefaultEnvironment="NO_PROXY={{GetNoProxy}}"
+    DefaultEnvironment="no_proxy={{GetNoProxy}}"
+    {{- end}}
+{{- end}}
+
+{{- if ShouldConfigureHTTPProxyCA}}
+- path: /usr/local/share/ca-certificates/proxyCA.pem
+  permissions: "0644"
```

Comment:
nit: encoding: base64

if we use base64 here then we we don't need to decode it in baker.go